src/routes/safe/components/Transactions/helpers/Simulation/simulation.ts

```diff
@@ -0,0 +1,212 @@
+  return `https://dashboard.tenderly.co/public/${TENDERLY_ORG_NAME}/${TENDERLY_PROJECT_NAME}/simulator/${simulationId}`
+}
+
+/* We need to overwrite the threshold stored in smart contract storage to 1
+ to do a proper simulation that takes transaction guards into account.
+ The threshold is stored in storage slot 4 and uses full 32 bytes slot
+ Safe storage layout can be found here:
+ https://github.com/gnosis/safe-contracts/blob/main/contracts/libraries/GnosisSafeStorage.sol */
+const THRESHOLD_ONE_STORAGE_OVERRIDE = {
+  [`0x${'4'.padStart(64, '0')}`]: `0x${'1'.padStart(64, '0')}`,
+}
+
+const getStateOverride = (
+  address: string,
+  balance?: string,
+  code?: string,
+  storage?: Record<string, string>,
+): Record<string, StateObject> => ({
+  [address]: {
+    balance,
+    code,
+    storage,
+  },
+})
+
+interface SafeTransaction {
+  to: string
+  value: string
+  data: string
+  safeTxGas: string
+  baseGas: string
+  gasPrice: string
+  gasToken: string
+  refundReceiver: string
+  nonce: string
+  operation: string
+}
+
+interface SignedSafeTransaction extends SafeTransaction {
+  signatures: string
+}
+
+const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000'
+
+const getBlockMaxGasLimit = async (web3: Web3): Promise<string> => {
+  const block = await web3.eth.getBlock('latest')
+  return block.gasLimit.toString()
+}
+
+const buildSafeTransaction = (template: OptionalExceptFor<SafeTransaction, 'to' | 'nonce'>): SafeTransaction => {
+  return {
+    to: template.to,
+    value: template.value || '0',
+    data: template.data || '0x',
+    operation: template.operation || '0',
+    safeTxGas: template.safeTxGas || '0',
+    baseGas: template.baseGas || '0',
+    gasPrice: template.gasPrice || '0',
+    gasToken: template.gasToken || ZERO_ADDRESS,
+    refundReceiver: template.refundReceiver || ZERO_ADDRESS,
+    nonce: template.nonce,
+  }
+}
+
+const encodeSafeExecuteTransactionCall = (tx: SignedSafeTransaction): string => {
+  const web3 = new Web3()
+
+  const encodedSafeExecuteTransactionCall = web3.eth.abi.encodeFunctionCall(
+    {
+      name: 'execTransaction',
+      type: 'function',
+      inputs: [
+        { name: 'to', type: 'address' },
+        { name: 'value', type: 'uint256' },
+        { name: 'data', type: 'bytes' },
+        { name: 'operation', type: 'uint8' },
+        { name: 'safeTxGas', type: 'uint256' },
+        { name: 'baseGas', type: 'uint256' },
+        { name: 'gasPrice', type: 'uint256' },
+        { name: 'gasToken', type: 'address' },
+        { name: 'refundReceiver', type: 'address' },
+        { name: 'signatures', type: 'bytes' },
+      ],
+    },
+    [
+      tx.to,
+      tx.value,
+      tx.data,
+      tx.operation,
+      tx.safeTxGas,
+      tx.baseGas,
+      tx.gasPrice,
+      tx.gasToken,
+      tx.refundReceiver,
+      tx.signatures,
+    ],
+  )
+
+  return encodedSafeExecuteTransactionCall
+}
+
+const getPreValidatedSignature = (address: string): string => {
```

Comment:
I believe we should have the same code somewhere in the repo already (should be used for approve & execute thing)